File: 백준/Silver/1920. 수 찾기/수 찾기.py
def binary_search(start,end,target):
    while start <= end :
        mid = (start + end) // 2

        if target == N_lst[mid]:
            return 1
        elif target > N_lst[mid]:
            start = mid + 1
        elif target < N_lst[mid]:
            end = mid - 1

    return 0

N = int(input())
N_lst = sorted(list(map(int,input().split())))
M = int(input())
M_lst = list(map(int,input().split()))

for m in M_lst :
    start, end = 0, N - 1
    print(binary_search(start, end, m))

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader( new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0 ; i < m; i++){
            int answer = 0;
            int start = 0, end = n - 1;
            int target = Integer.parseInt(st.nextToken());

            while (start <= end){
                int mid = (start + end) / 2;
                if(arr[mid] == target) {
                    answer = 1;
                    break;
                }
                if (arr[mid] > target) end = mid - 1;
                else  start = mid + 1;
            }
            System.out.println(answer);
        }
    }
}
